cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -169,38 +192,21 @@ func TestNodeGroupNewNodeGroup(t *testing.T) {
 			t.Errorf("expected %t, got %t", false, result)
 		}
 
-		nodes, err := ng.Nodes()
-		if err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
-
-		if len(nodes) != tc.nodeCount {
-			t.Errorf("expected %d nodes, got %v", tc.nodeCount, len(nodes))
-		}
+		// We test ng.Nodes() in TestControllerNodeGroupsNodeCount
 	}
 
 	t.Run("MachineSet", func(t *testing.T) {
-		for i, tc := range testCases {
+		for _, tc := range testCases {
 			t.Run(tc.description, func(t *testing.T) {
-				testObjs := newMachineSetTestObjs(t.Name(), i, tc.nodeCount, tc.replicas, tc.annotations)
-				tc.namespace = testObjs.spec.namespace
-				tc.name = fmt.Sprintf("%s%d", testObjs.spec.machineSetPrefix, i)
-				tc.id = path.Join(tc.namespace, tc.name)
-				tc.debug = fmt.Sprintf("%s (min: %d, max: %d, replicas: %d)", path.Join(tc.namespace, tc.name), tc.minSize, tc.maxSize, tc.replicas)
-				testNodeGroupProperties(t, tc, testObjs)
+				test(t, tc, mustCreateMachineSetTestConfig(testNamespace, tc.nodeCount, tc.replicas, tc.annotations))
```

Comment:
Good, we don't care how a machineset is created as long as the MS object is a valid object.